src/main/java/com/suse/pase/query/QueryFactory.java

```diff
@@ -40,6 +38,23 @@
             .collect(toList());
     }
 
+    /** Parses a patch from an input stream and returns queries (one per file) for the applied context of it **/
+    public static List<PatchTargetQuery> buildAppliedPatchTargetQuery(InputStream is) throws IOException {
+        var ud = UnifiedDiffReader.parseUnifiedDiff(is);
+
+        return ud.getFiles().stream()
+                .filter(f -> !f.getFromFile().equals("/dev/null"))
+                .map(file ->{
+                    var path = file.getFromFile();
+                    var chunks = file.getPatch().getDeltas().stream()
+                            .filter(d -> of(INSERT, EQUAL, CHANGE).contains(d.getType()))
+                            .map(d -> d.getTarget().getLines().stream().filter(l -> !l.isEmpty()).collect(toList()))
+                            .collect(toList());
+                    return new PatchTargetQuery(path, chunks);
```

Comment:
```suggestion
                    return new PatchQuery(path, chunks);
```

Alternate class name suggestion, to make clear there are two possible cases for this query (patch target or applied patch).

PS. I recommend to use an IDE to do this kind of refactoring (renaming) in a safe way.

I highly recomment IntelliJ IDEA if you are not using it already.

Reply:
Once the class is renamed to `PatchQuery`, I'd DRY up `buildAppliedPatchTargetQuery` and `buildPatchTargetQuery` by creating a generic `buildPatchQuery` which takes a `List<DeltaType>` as parameter.

Then I can imagine implementing `buildAppliedPatchTargetQuery` by calling `buildPatchQuery` with `of(DELETE, EQUAL, CHANGE)` and `buildAppliedPatchTargetQuery` with `of(INSERT, EQUAL, CHANGE)`.

Finally I would recommend renaming `buildAppliedPatchTargetQuery` to `buildAppliedPatchQuery` as "target" does not make too much (English) sense at that point.